### PR TITLE
VR Games Night 👾 - Sofia Vaz Sousa 

### PR DESCRIPTION
**Pull request by sofiavazs**
**Link to my deployed project:**
[VR Games Night! ](https://vrgamesnight.netlify.app/)

**My reflections on how this week's project turned out:**

I'm really proud how this week's project turned out, I did a thorough description on my readme how I planned it and what I learned.

It was challenging but even though React is so new, I feel **weirdly** comfortable with it? which is a really alien feeling for me, since I always felt not very confident with Javascript, I think it probably has to do with building components and breaking down a big block of code into smaller chunks, which makes it easier to understand, however things get a bit confusing with so many components but I feel I got a good grasp of the state hook and how to pass and store data between components and that feels really good.

Obviously I'm not saying with this that I understand the "ins" and "outs" of React, because I don't 😅  it just feels a little better than when I started with JavaScript 😊
I would have made it responsive if I had more time, but I still plan on doing it

**Things I'd like to have clarified or explained in more detail:**
This week was actually good regarding the materials and I asked a lot of questions, so I don't feel I need anything clarified as of now 😊

### CODE REVIEW

File: code/src/Form.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    <>
+    <h1 tabIndex='0'>VR Games Night!</h1>
+    <section className='form-container'>
+        {section !== 'summary' ? (
+            <form onSubmit={handleSubmit}>
+                {section === 'welcomePage' && (
+                    <WelcomePage/>   
+                )}
+                {section === 'nameQuestion' && (
+                    <TextInput name= {name} setName= {setName}/>
```
Comment (HenrikeW) [inline code/src/Form.js:48]: Now I'm being picky, but I think you usually don't make a space after = when writing props, same in the lines below.

File: code/src/index.css [hunk 2]
```diff
@@ -5,9 +5,212 @@ body {
     sans-serif;
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
+  background: linear-gradient( rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3) ), url('./assets/sparc-bg.jpg');
```
Comment (HenrikeW) [inline code/src/index.css:8]: I don't understand why you have a gradient with two times the same color.

File: code/src/index.css [hunk 3]
```diff
@@ -5,9 +5,212 @@ body {
...
+  background-size: cover;
+  
 }
 
 code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
+
+/* header and header animation  */
```
Comment (HenrikeW) [inline code/src/index.css:18]: I'm super impressed by this whole header and animation thing!! 💯

File: code/src/index.css [hunk 4]
```diff
@@ -5,9 +5,212 @@ body {
...
+  margin: 30px;
+}
+
+form {
+  display:flex;
+  flex-direction: column;
+  padding-top: 15px;
+  padding-bottom: 15px;
+  padding-right:30px;
+  padding-left: 30px;
```
Comment (HenrikeW) [inline code/src/index.css:135]: I learned from Maks in one of his last code reviews that you always start with "top" and then go round like the clock, so top, left, bottom, right.

File: code/src/index.css [hunk 5]
```diff
@@ -5,9 +5,212 @@ body {
...
+input[type=radio], select {
+  margin-right: 10px;
+}
+
+input:valid {
+  border-color:#81b214;
+}
+
+input:invalid {
+  border-color: red;
```
Comment (HenrikeW) [inline code/src/index.css:167]: Great that you have your own visualization of the validation!

File: code/src/index.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
-import './index.css'
+
 import { App } from './App'
+import './index.css'
```
Comment (HenrikeW) [inline code/src/index.js:5]: Very good to move it here 👍

File: code/src/Components/RadioButtons.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+const gamesList = [
+    'Beat Saber',
+    'SPARC',
+    'Until you Fall',
+    'Pistol Whip',
+    'No Man\'s Sky',
+    'Half-Life: Alyx'
+]
+export const RadioButtons = ({game, setGamesList}) => {
```
Comment (HenrikeW) [inline code/src/Components/RadioButtons.js:11]: For some reason, when tab-ing through your survey it took many tabs to reach the radio buttons and their options. I ended up on the Next-button first and then in a lot of menus in my browser before I got to the radio buttons while only tabing forward. Unfortunately, I don't know if there is anything you can do about it :/

**Review by HenrikeW — APPROVED** [review]
You did a great job on this week's project, Sofia! You fulfilled the blue level, parts of the red AND even parts of the black one, super cool! The solution to make it multi-step is easy to understand and your comments in the code are very helpful. And I love your animation on the welcome page <3

I commented some things, just have a look at it. There are a few parts where you could use more semanic options if you want to improve your code. (But of course, as you used aria-labels it's accessible anyway). 

The only thing I found a bit confusing was your naming of the component files because you partly refer to the content ("snacks") and partly to the type of input, but then one time it's "SelectDropdown" and one time it's just "Dropdown"... :D For me seeing the files for the first time it took a while to grab what was there.

Other than that - keep up the great work!!

File: code/src/Components/RadioButtons.js (hunk 7 above)
Comment (sofiavazs) [reply code/src/Components/RadioButtons.js:11]: I did have an issue with the `focus` and solved it by having `autoFocus` but don't know why doesn't work in here to be honest :\

File: code/src/index.css (hunk 4 above)
Comment (sofiavazs) [reply code/src/index.css:135]: nice, missed that! thank you!

File: code/src/index.css (hunk 2 above)
Comment (sofiavazs) [reply code/src/index.css:8]: Did try to remove one of them and my image disappears, didn't realise that before :\

File: code/src/Form.js (hunk 1 above)
Comment (sofiavazs) [reply code/src/Form.js:48]: Not picky at all, it was my mistake! thank you

File: code/src/Components/Summary.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      imageSrc = NoMansSkyImage;
+      break;
+    case 'Half-Life: Alyx': 
+      imageSrc = HalfLifeAlyxImage;
+      break;
+  default:
+    break;
+  }
+
+  return <img src={imageSrc} className= 'summary-image' alt='game beat saber'/>
```
[the remark this answers is not in the document]
Comment (sofiavazs) [reply code/src/Components/Summary.js:41]: yes.. no idea why i left it like that! thank you!

File: code/src/App.js [hunk 9]
```diff
@@ -1,9 +1,16 @@
 import React from 'react'
 
+import {Form} from './Form'
+import {Footer} from './Footer'
```
[the remark this answers is not in the document]
Comment (sofiavazs) [reply code/src/App.js:4]: Yeah, since react that I get a little lost where to place the ;

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Really nice work Sofia! Changing the status to CR though until you fix the inconsistent semicolons - they should be everywhere or nowhere :)

**Comment by sofiavazs** [comment]
> Really nice work Sofia! Changing the status to CR though until you fix the inconsistent semicolons - they should be everywhere or nowhere :)

Thank you Matilda, I did change it straight away (you can see my commit)but for some reason is not showing outdated here, I'll have a look!

**Comment by sofiavazs** [comment]
> Really nice work Sofia! Changing the status to CR though until you fix the inconsistent semicolons - they should be everywhere or nowhere :)

Thank you Matilda! Turned out I added the semicolons, but missed the app.js, I believe they're now EVERYWHERE 😅

**Comment by HIPPIEKICK** [comment]
> Thank you Matilda! Turned out I added the semicolons, but missed the app.js, I believe they're now EVERYWHERE 😅

Great! Looks like your return statements are still missing semicolons though :)

**Comment by sofiavazs** [comment]
Thanks Matilda, embarrassing... it's so easy to miss them or don't know where to put them, now that the projects are getting more complex... Hopefully I haven't missed any this time 🙈

**Comment by HIPPIEKICK** [comment]
> Thanks Matilda, embarrassing... it's so easy to miss them or don't know where to put them, now that the projects are getting more complex... Hopefully I haven't missed any this time 🙈

Not embarrassing - super good practice! I know it's hard but it's super important because a lot of recruiters look at these things like consistency, indentation etc. Looks good now so approving ⭐

**Review by HIPPIEKICK — APPROVED** [review]

**Pull request closed without merging**